packages/adapter-node/index.js

```diff
@@ -111,3 +114,26 @@ export default function (opts = {}) {
 		}
 	};
 }
+
+/**
+ * Load sourcemaps for files in the tmp directory so that the final ones
+ * point to the original source files, instead of the generated files in outDir.
+ * @param {string} tmp
+ * @returns {import('rollup').Plugin}
+ */
+function merge_sourcemap_plugin(tmp) {
+	const should_process_sourcemaps = createFilter(`${normalizePath(tmp)}/**/*.js`);
+
+	return {
+		name: 'adapter-node-sourcemap-loader',
+		async load(id) {
+			if (!should_process_sourcemaps(id)) return;
+			if (!existsSync(`${id}.map`)) return;
+			const [code, map] = await Promise.all([
+				readFile(id, 'utf-8'),
```

Comment:
why is this using `readFile` when other places use `readFileSync`? Is this really an observable performance difference?

Reply:
this one is reading two files in parallel, so I think `readFile` would be more performant here, so if we want to standardize I think I'd rather use `readFile` than `readFileSync`